app/page.tsx

```diff
@@ -52,66 +55,90 @@ export default function Home() {
     const provider = new ethers.BrowserProvider(window.ethereum);
     const setContractVal = async () => {
       try {
-        const signer = await provider.getSigner();
-        const contract = new ethers.Contract(contractAddress, NFT.abi, signer);
-        const address = await signer.getAddress();
+        const signer: ethers.JsonRpcSigner = await provider.getSigner();
+        const contract = new ethers.Contract(
+          contractAddress,
+          NFT.abi,
+          signer
+        )as BaseContract as SuperCarERC1155;;
+        const address: string = await signer.getAddress();
+        const NFTContract = SuperCarERC1155__factory.connect(
+          contractAddress,
+          provider
+        );
+
         setAccount(address);
         setContract(contract);
 
         if (contract) {
-          const cMintPrice = await contract.currentPrice();
-          const nftB = await contract.balanceOf(await signer.getAddress(), 0);
-          setMintPrice(ethers.formatEther(Number(cMintPrice).toString()));
-          setBurnPrice(localStorage.getItem("burnPrice"));
-          setNftBalance(Number(nftB));
+          const cMintPrice: number = Number(await contract.currentPrice());
+          const nftB: number = Number(
+            await contract.balanceOf(await signer.getAddress(), 0)
+          );
+          setMintPrice(Number(ethers.formatEther(cMintPrice.toString())));
```

Comment:
cMintPrice is already number why we are converting it into string

Reply:
formatEther takes string as argument